### PR TITLE
B 17926: New download orders Prime api  for ghc counseling

### PR DESCRIPTION
**Pull request by MInthavongsay**
## Summary

The objective of this PR is to create a new Prime API ([E-04799](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A864544)) to allow consumers to download all uploaded move orders in one merged PDF file.  This PR only contains stubbed response. There is a down stream dependency to develop the service layer to generate the PDF payload - [B-18027](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A870406).  Complete implementation of API will be done when B-18027 is completed.  For now it will just return empty PDF file with routine validations (HTTP Status Codes: 400, 403, 404, 422, 500).

**GET:  /moves/{locator}/order/download**

Business Rule Validations:

- Move must be in Needs Service Counseling state 

- Move owner/client's origin duty location does not provide services counseling 

Service will return 422 (UnprocessableEntity) status code if any of the business validations fail.

## Changes

- Updated: prime/api.go,  swagger/prime.yaml
- New Prime handler and supporting code
- Added new unit tests
- **Updated .circleci/config.yml, swagger-def/.spectral.yml. (resolve CircleCI Spectral tests failures)**

## Verification Steps for Reviewers
- Go to http://primelocal:3000/swagger-ui/prime.html#/moveTaskOrder/downloadMoveOrder and execute API
- Run unit tests: pkg/handlers/primeapi/move_task_order_test.go -- TestDownloadMoveOrderHandle() test suite

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

- Go to http://primelocal:3000/swagger-ui/prime.html#/moveTaskOrder/downloadMoveOrder and execute API
- Use move locator code in "needs service counseling" state and origin duty location does not have gov services counseling. ((origin)duty_locations.provides_services_counseling must be FALSE)
`SELECT m.locator
FROM   (SELECT o.*
        FROM   PUBLIC.orders o
               JOIN PUBLIC.duty_locations dl
                 ON o.origin_duty_location_id = dl.id
                    AND dl.provides_services_counseling = false) t
       JOIN PUBLIC.moves m
         ON t.id = m.orders_id
            AND m.status = 'NEEDS SERVICE COUNSELING';  `

### Frontend

- [ ] http://primelocal:3000/swagger-ui/prime.html#/moveTaskOrder/downloadMoveOrder

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: pkg/handlers/primeapi/move_task_order.go [hunk 1]
```diff
@@ -194,3 +198,73 @@ func (h UpdateMTOPostCounselingInformationHandler) Handle(params movetaskorderop
...
+					unprocessableErr := apperror.NewUnprocessableEntityError(errMessage)
+					appCtx.Logger().Info(unprocessableErr.Error())
+					payload := payloads.ValidationError(unprocessableErr.Error(), h.GetTraceIDFromRequest(params.HTTPRequest), nil)
+					return movetaskorderops.NewDownloadMoveOrderUnprocessableEntity().
+						WithPayload(payload), unprocessableErr
+				}
+			}
+
+			// TODO: wire up PDF service to generate real PDF file. For now return empty PDF file.
+			// https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A864544
```
Comment (danieljordan-caci) [inline pkg/handlers/primeapi/move_task_order.go:263]: Can you expand on this TODO? Is this going to be addressed in a later feature or BL item?

I pulled up the link - would B-18027 address this?

Comment (MInthavongsay) [reply pkg/handlers/primeapi/move_task_order.go]: My bad. Copy/paste wrong url.

**Review by danieljordan-caci — APPROVED** [review]
lgtm. 

Tests are passing, logic looks sound, comments are appreciated.

Also appreciate you not including excessive response codes in your swagger add-ins that aren't handled in your handler.

**Comment by r-mettler** [comment]
Tests pass and swagger calls work.

**Review by deandreJones — APPROVED** [review]

File: pkg/handlers/primeapi/move_task_order_test.go [hunk 2]
```diff
@@ -1842,3 +1843,211 @@ func (suite *HandlerSuite) TestUpdateMTOPostCounselingInfo() {
...
 }
+
+func (suite *HandlerSuite) TestDownloadMoveOrderHandler() {
+	suite.Run("Successful DownloadMoveOrder - 200", func() {
+		mockMoveSearcher := mocks.MoveSearcher{}
+		mockOrderFetcher := mocks.OrderFetcher{}
+
+		move := factory.BuildNeedsServiceCounselingMove(suite.DB(), nil, nil)
+
+		// Hardcode to true to indicate duty location does not provide GOV counseling
```
Comment (r-mettler) [inline pkg/handlers/primeapi/move_task_order_test.go:1854]: Nit pick: You say hardcode to true but you hardcoded to false. The comment should be changed to match or just say Harcode to indicate...

**Review by r-mettler — APPROVED** [review]

**Pull request merged**